dev/io.openliberty.microprofile.health.3.1.internal_fat/fat/src/io/openliberty/microprofile/health31/fat/FailedApplicationStartTest.java

```diff
@@ -0,0 +1,241 @@
+        }
+
+        // Check /health/ready -> expect that we report DOWN.
+        connhealth = HttpUtils.getHttpConnectionWithAnyResponseCode(server, READY_ENDPOINT);
+        jsonResponse = getJSONPayload(connhealth);
+        checks = (JsonArray) jsonResponse.get("checks");
+        assertTrue("The JSON response was not empty.", checks.isEmpty());
+        assertEquals("The status of the ready health check was epected to be DOWN, but we received UP.", jsonResponse.getString("status"), "DOWN");
+
+        // Check /health/live -> expect that we report UP.
+        connhealth = HttpUtils.getHttpConnectionWithAnyResponseCode(server, LIVE_ENDPOINT);
+        jsonResponse = getJSONPayload(connhealth);
+        checks = (JsonArray) jsonResponse.get("checks");
+        assertTrue("The JSON response was not empty.", checks.isEmpty());
+        assertEquals("The status of the live health check was epected to be UP, but we received DOWN.", jsonResponse.getString("status"), "UP");
+
+    }
+
+    /**
+     * Test that loads an app that throws a java.lang.Error in the servlet container intializer.
+     * We expect to see a CWWKZ0012I because of it and that the health check (overall health check) reports DOWN.
+     *
+     * This test tests the app in apps folder.
+     *
+     * @throws Exception
+     */
+    @ExpectedFFDC("java.lang.Error")
+    @Test
+    public void webModuleStartsandFailsTest_apps() throws Exception {
+
+        server = serverAppDir;
+
+        WebArchive app = ShrinkHelper.buildDefaultApp(APP_NAME, "io.openliberty.microprofile.health31.start.and.fails.app")
+                        .addAsServiceProvider(ServletContainerInitializer.class.getName(),
+                                              "io.openliberty.microprofile.health31.start.and.fails.app.AppServletContainerInitializer");
+
+        ShrinkHelper.exportAppToServer(server, app, DeployOptions.DISABLE_VALIDATION, DeployOptions.SERVER_ONLY);
+
+        server.startServer();
+
+        log("webModuleStartsandFailsTest", "Checking (and waiting) for CWWKZ0012I");
+        String ret = server.waitForStringInLog("CWWKZ0012I");
+        assertNotNull("We expected to find CWWKZ0012I, but was not found", ret);
+
+        // Check /health -> expect that we report DOWN.
+        HttpURLConnection connhealth = HttpUtils.getHttpConnectionWithAnyResponseCode(server, HEALTH_ENDPOINT);
+        JsonObject jsonResponse = getJSONPayload(connhealth);
+        JsonArray checks = (JsonArray) jsonResponse.get("checks");
+        assertTrue("The JSON response was not empty.", checks.isEmpty());
+        assertEquals("The status of the (overall) health check was epected to be DOWN, but we received UP.", jsonResponse.getString("status"), "DOWN");
+
+        // Check /health/started -> expect that we report DOWN.
+        // Must skip for mpHealth-2.0, does not contain start health check
```

Comment:
Include mpHealth-3.0 in the comments as well.
```suggestion
        // Must skip for mpHealth-2.0 and mpHealth-3.0, does not contain start health check
```